apps/gateway/src/chat/chat.ts

```diff
@@ -287,24 +290,84 @@ chat.openapi(completions, async (c) => {
 
 	let url: string | undefined;
 
-	// Get the provider key for the selected provider
-	const providerKey = await db.query.providerKey.findFirst({
-		where: {
-			status: {
-				eq: "active",
-			},
-			projectId: {
-				eq: apiKey.projectId,
-			},
-			provider: {
-				eq: usedProvider,
-			},
-		},
-	});
+	// Get the provider key for the selected provider based on project mode
+	const project = await getProject(apiKey.projectId);
+
+	if (!project) {
+		throw new HTTPException(500, {
+			message: "Could not find project",
+		});
+	}
+
+	let providerKey;
+
+	if (project.mode === "api-keys") {
+		// Get the provider key from the database using cached helper function
+		providerKey = await getProviderKey(apiKey.projectId, usedProvider);
+
+		if (!providerKey) {
+			throw new HTTPException(400, {
+				message: `No API key set for provider: ${usedProvider}. Please add a provider key in your settings.`,
+			});
+		}
+	} else if (project.mode === "credits") {
+		// Check if the organization has enough credits using cached helper function
+		const organization = await getOrganization(project.organizationId);
+
+		if (!organization) {
+			throw new HTTPException(500, {
+				message: "Could not find organization",
+			});
+		}
+
+		if (organization.credits <= 0) {
+			throw new HTTPException(402, {
+				message: "Organization has insufficient credits",
+			});
+		}
+
+		let token: string | undefined;
+
+		switch (usedProvider) {
+			case "openai":
+				token = process.env.OPENAI_API_KEY;
+				break;
+			case "anthropic":
+				token = process.env.ANTHROPIC_API_KEY;
+				break;
+			case "google-vertex":
+				token = process.env.VERTEX_API_KEY;
+				break;
+			case "inference.net":
+				token = process.env.INFERENCE_API_KEY;
+				break;
+			case "kluster.ai":
+				token = process.env.KLUSTER_API_KEY;
+				break;
+			default:
+				throw new HTTPException(400, {
+					message: `No environment variable set for provider: ${usedProvider}`,
+				});
+		}
 
-	if (!providerKey) {
+		if (!token) {
+			throw new HTTPException(400, {
+				message: `No API key set in environment for provider: ${usedProvider}`,
+			});
+		}
+
+		providerKey = {
+			id: `env-${usedProvider}`,
+			token,
+			provider: usedProvider,
+			status: "active",
+			projectId: apiKey.projectId,
+			createdAt: new Date(),
+			updatedAt: new Date(),
+		};
```

Comment:
do not create this here, just handle if providerKey is not set below. adapt schema.ts / relations.ts if necessary as with this PR, there will be no related provider key entry in the DB